File: webapp/greetings.js
// Personalized greeting pool for Rudhran.
// Indexed by time-of-day buckets and mood. Picked deterministically by date so
// the same morning shows the same greeting until refresh, but you'll see new
// ones across the week — closer to how Claude's home greetings cycle.

window.HARRY_GREETINGS = (function () {
  const NAME = "Rudhran";

  const dawn = [
    `Good morning, ${NAME}.`,
    `Up before the sun, ${NAME}?`,
    `Quiet hours, ${NAME}. The world's still asleep.`,
    `Early start. I admire it, ${NAME}.`,
    `The kettle's metaphorical, but consider it on.`,
    `First light. Where shall we begin, ${NAME}?`,
    `Morning, ${NAME}. Inbox quiet. Coast clear.`,
  ];

  const morning = [
    `Good morning, ${NAME}.`,
    `Ready when you are, ${NAME}.`,
    `Morning, ${NAME}. What's on the docket?`,
    `Welcome back, ${NAME}.`,
    `At your service, ${NAME}.`,
    `Pleasure to see you again, ${NAME}.`,
    `Top of the morning, ${NAME}.`,
    `${NAME}. A fine day for building things.`,
    `Coffee or cold start, ${NAME}?`,
    `Morning. Eight new commits since you left.`,
    `Standing by, ${NAME}.`,
    `${NAME}, you have my attention.`,
  ];

  const midday = [
    `Welcome back, ${NAME}.`,
    `Hello again, ${NAME}.`,
    `Afternoon, ${NAME}. Ready to continue?`,
    `${NAME}. Where were we?`,
    `Picking up where we left off, ${NAME}?`,
    `${NAME}, the floor is yours.`,
    `Right on time, ${NAME}.`,
    `${NAME}, ready when you are.`,
    `${NAME}. Shall we?`,
    `Good afternoon, ${NAME}.`,
    `Listening, ${NAME}.`,
  ];

  const evening = [
    `Good evening, ${NAME}.`,
    `Evening, ${NAME}. Long day?`,
    `Welcome back, ${NAME}.`,
    `${NAME}. The lights are low. So am I.`,
    `Easy does it, ${NAME}.`,
    `Evening, ${NAME}. What's on your mind?`,
    `${NAME}, dinner can wait. What do you need?`,
    `Sunset shift, ${NAME}. I'm all yours.`,
    `${NAME}, let's wrap the day cleanly.`,
    `Good evening. Two unread, one urgent.`,
  ];

  const night = [
    `Burning the midnight oil, ${NAME}?`,
    `Late hours suit you, ${NAME}.`,
    `${NAME}. Quiet on the western front.`,
    `Still here, ${NAME}. Always.`,
    `Night shift, ${NAME}. What can I help with?`,
    `${NAME}, the city's asleep. We don't have to be.`,
    `Welcome back, ${NAME}. Don't tell anyone I noticed the hour.`,
    `${NAME}. Whisper if you'd like.`,
    `Standing watch, ${NAME}.`,
    `${NAME}, I'll be brief — and so should you, perhaps.`,
  ];

  // Mood / context flavors — drawn occasionally regardless of time
  const flavor = [
    `${NAME}. Let's make something good.`,
    `Ready to think clearly, ${NAME}?`,
    `${NAME}, the world is small and the work is large. Begin.`,
    `One step at a time, ${NAME}.`,
    `${NAME}, your move.`,
    `I've been thinking, ${NAME}.`,
    `${NAME}. Less talk, more progress?`,
    `Curious what we'll build today, ${NAME}.`,
    `${NAME}, the boring parts are handled. Focus on the interesting ones.`,
    `What needs solving, ${NAME}?`,
    `${NAME}. Tell me what good looks like.`,
    `${NAME}, the agenda is yours.`,
    `Brief and bright, ${NAME} — or long and slow. Your call.`,
    `${NAME}. I've been quiet. Productively so.`,
    `One small task or a tall mountain, ${NAME}?`,
  ];

  // Returning user / streak vibes
  const familiar = [
    `Back so soon, ${NAME}?`,
    `${NAME}. Round two.`,
    `Didn't expect you yet, ${NAME}. Pleasant surprise.`,
    `${NAME}. Right where we left off.`,
    `${NAME}, the kettle's still on, metaphorically.`,
  ];

  function bucket(hour) {
    if (hour >= 4 && hour < 7) return "dawn";
    if (hour >= 7 && hour < 12) return "morning";
    if (hour >= 12 && hour < 17) return "midday";
    if (hour >= 17 && hour < 21) return "evening";
    return "night";
  }

  const pools = { dawn, morning, midday, evening, night, flavor, familiar };

  // Build a flat catalog (deduped) for browsing in settings — "memory" feel
  const all = [];
  for (const k of Object.keys(pools)) {
    for (const g of pools[k]) {
      if (!all.includes(g)) all.push(g);
    }
  }

  function pick(seed) {
    const d = new Date();
    const hr = d.getHours();
    const b = bucket(hr);
    // Mix 80% time-of-day pool, 15% flavor, 5% familiar
    const r = ((seed ?? Math.random()) * 100) | 0;
    let pool;
    if (r < 80) pool = pools[b];
    else if (r < 95) pool = flavor;
    else pool = familiar;
    return pool[Math.floor(Math.random() * pool.length)];
  }

  return { pick, all, pools, NAME, bucket };
})();

File: harry/greetings.py
"""Python mirror of the JS greeting pools. Used by the backend to speak
the opening greeting through the speaker as soon as the app launches.

The pools are intentionally shorter than the JS catalogues — the JS side
remains the authoritative visible catalogue; this is the spoken subset."""
from __future__ import annotations

import os
import random
from datetime import datetime

EN = {
    "dawn": [
        "Good morning, {name}.",
        "Up before the sun, {name}.",
        "Early start, {name}. The world is still asleep.",
    ],
    "morning": [
        "Good morning, {name}.",
        "Welcome back, {name}.",
        "Ready when you are, {name}.",
        "Top of the morning, {name}.",
    ],
    "midday": [
        "Welcome back, {name}.",
        "Afternoon, {name}. Ready to continue?",
        "Right on time, {name}.",
    ],
    "evening": [
        "Good evening, {name}.",
        "Evening, {name}. Long day?",
        "{name}, let us wrap the day cleanly.",
    ],
    "night": [
        "Burning the midnight oil, {name}?",
        "Late hours suit you, {name}.",
        "{name}. The city is asleep. We do not have to be.",
        "Standing watch, {name}.",
    ],
}

TA = {
    "dawn": [
        "காலை வணக்கம், {name}.",
        "சூரியனுக்கு முன்பே எழுந்துவிட்டீர்களா, {name}?",
    ],
    "morning": [
        "காலை வணக்கம், {name}.",
        "மீண்டும் வருக, {name}.",
        "தயாராக உள்ளேன், {name}.",
    ],
    "midday": [
        "மீண்டும் வருக, {name}.",
        "மதிய வணக்கம், {name}.",
    ],
    "evening": [
        "மாலை வணக்கம், {name}.",
        "{name}, இந்த நாளை நேர்த்தியாக முடிப்போம்.",
    ],
    "night": [
        "நள்ளிரவு எண்ணெய் எரிக்கிறீர்களா, {name}?",
        "{name}, நகரம் தூங்குகிறது. நாம் தூங்க வேண்டாம்.",
        "{name}, நான் இங்கேயே இருக்கிறேன். எப்போதும்.",
    ],
}


def _bucket(hour: int) -> str:
    if 4 <= hour < 7:  return "dawn"
    if 7 <= hour < 12: return "morning"
    if 12 <= hour < 17: return "midday"
    if 17 <= hour < 21: return "evening"
    return "night"


TA_NAME_BY_EN = {"Rudhran": "ருத்ரன்"}


def pick_greeting(language: str = "en", name: str | None = None) -> str:
    name = (name or os.getenv("HARRY_USER_NAME", "Rudhran")).strip() or "Rudhran"
    hour = datetime.now().hour
    bucket = _bucket(hour)
    use_ta = language == "ta" or (language == "auto" and hour % 2)
    pool = TA[bucket] if use_ta else EN[bucket]
    spoken = (os.getenv("HARRY_USER_NAME_TA") or TA_NAME_BY_EN.get(name, name)) \
        if use_ta else name
    return random.choice(pool).format(name=spoken)
